salt/modules/state.py

```diff
@@ -98,17 +98,16 @@ def _set_retcode(ret):
         __context__['retcode'] = 2
 
 
-def _check_pillar(kwargs, pillar=None):
+def _get_pillar_errors(kwargs, pillar=None):
     '''
-    Check the pillar for errors, refuse to run the state if there are errors
-    in the pillar and return the pillar errors
+    Checks all pillars (external and internal) for errors.
+    Return an error message, if anywhere or None.
+
+    :param kwargs: dictionary of options
+    :param pillar: external pillar
+    :return: None or an error message
     '''
-    if kwargs.get('force'):
-        return True
-    pillar_dict = pillar if pillar is not None else __pillar__
-    if '_errors' in pillar_dict:
-        return False
-    return True
+    return None if kwargs.get('force') else (pillar or {}).get('_errors', __pillar__.get('_errors')) or None
```

Comment:
I would probably unpack this one-line into:

```python
if kwargs.get('force'):
    return None
_errors = (pillar or {}).get('_errors', __pillar__.get('_errors')
return _errors if _errors else None
```

That would make more intuitive to get the actual behavior of this function. I'm not a big fan of compressing too much logic into one line as it might lead into confusion.

> Sparse is better than dense. (https://www.python.org/dev/peps/pep-0020/)

Reply:
I don't like several returns. Besides this is just a port from what is in 2017+